tests/e2escenarios/e2e-test.go

```diff
@@ -0,0 +1,258 @@
+			deploymentName := "my-component"
+			serviceName := "my-cs"
+			getDeployArgs := []string{"get", "deployment", "-n", commonVar.Project}
+			getSVCArgs := []string{"get", "svc", "-n", commonVar.Project}
+
+			command := []string{"odo", "init"}
+			output, err := helper.RunInteractive(command, nil, func(ctx helper.InteractiveContext) {
+
+				helper.ExpectString(ctx, "Select language")
+				helper.SendLine(ctx, "javascript")
+
+				helper.ExpectString(ctx, "Select project type")
+				helper.SendLine(ctx, "Node.js\n")
+
+				helper.ExpectString(ctx, "Which starter project do you want to use")
+				helper.SendLine(ctx, "nodejs-starter\n")
+
+				helper.ExpectString(ctx, "Enter component name")
+				helper.SendLine(ctx, componentName)
+
+				helper.ExpectString(ctx, "Your new component '"+componentName+"' is ready in the current directory")
+
+			})
+			Expect(err).To(BeNil())
+			Expect(output).To(ContainSubstring("Your new component '" + componentName + "' is ready in the current directory"))
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElement("devfile.yaml"))
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElement("server.js"))
+
+			// "execute odo dev and add changes to application"
+			var devSession helper.DevSession
+			var outContents []byte
+			var ports map[string]string
+
+			devSession, outContents, _, ports, err = helper.StartDevMode()
+			helper.ReplaceString(filepath.Join(commonVar.Context, "server.js"), "from Node.js", "from updated Node.js")
+			Expect(err).ToNot(HaveOccurred())
+			helper.MatchAllInOutput(string(outContents), []string{"Watching for changes in the current directory"})
+
+			// "should update the changes"
+			checkIfDevEnvIsUp(ports["3000"], "Hello from updated Node.js Starter Application!")
+
+			// "changes are made made to the applications"
+			helper.ReplaceString(filepath.Join(commonVar.Context, "server.js"), "from updated Node.js", "from Node.js app v2")
+			helper.MatchAllInOutput(string(outContents), []string{"Watching for changes in the current directory"})
+
+			// "should deploy new changes"
+			checkIfDevEnvIsUp(ports["3000"], "Hello from Node.js app v2 Starter Application!")
+
+			// "running odo list"
+			stdout := helper.Cmd("odo", "list").ShouldPass().Out()
+			helper.MatchAllInOutput(stdout, []string{componentName, "nodejs", "Dev"})
+
+			// "exit dev mode and run odo deploy"
+			devSession.Stop()
+			devSession.WaitEnd()
+
+			// "run odo deploy"
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-deploy.yaml"), path.Join(commonVar.Context, "devfile.yaml"))
+			helper.ReplaceString(filepath.Join(commonVar.Context, "devfile.yaml"), "nodejs-prj1-api-abhz", componentName)
+
+			//helper.ReplaceString(filepath.Join(commonVar.Context, "devfile.yaml"), "quay.io/unknown-account/myimage", "quay.io/repository/libpod/alpine")
+			stdout = helper.Cmd("odo", "deploy").AddEnv("PODMAN_CMD=echo").ShouldPass().Out()
```

Comment:
It could be interesting to build the image for the e2e tests

Reply:
Not for this PR, but we should discuss this during a cabal

Reply:
I will create an issue for this one separately to track.